de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/structure/VirSatProjectResourceChangeListener.java

```diff
@@ -122,10 +122,12 @@ public boolean visit(IResourceDelta delta) {
 			// workspace job as we will need the editing domain to handle resource changes.
 			// This way the current job calling us can safely finish and the new job
 			// will handle the actual changes.
-			Job job = new WorkspaceSynchronizerJob(addedDvlmResources, removedDvlmResources, changedDvlmResources, counter++);
-			String changeList = printLists(addedDvlmResources, removedDvlmResources, changedDvlmResources);
-			Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VirSatProjectResourceChangeListener: Scheduling " + job.getName() + ". \n" + changeList));
-			job.schedule();
+			if (!addedDvlmResources.isEmpty() || !removedDvlmResources.isEmpty() || !changedDvlmResources.isEmpty()) {

```

Comment:
This is also a dificult one. The whole story to the workspace synchtronizer is, to detect if there are changes to a reosurce of the reosurceset which have been introduced somewhere on the workspace. In that case, the reosurce is out of sync with the loaded reosurce. EMF usually handles this in the generated editors. In VirSat we decided to handle this centrally, to also manage intended changes. E.g. when a resource is explicitly saved, which creates a workspace resource change event as well. These intended changes should obviously not lead to a resource reload. So at least we need to have a convention, agreement how to deal with non dvlm resources. In this proposed implementation, specific editors have to deal with reloading the resources.

Reply:
I have no problem at all iwth the WorkspaceSynchronizer. I actually like it :) I just want to make sure that is only triggered if there is actually a relvant change on the project. With the current implementation the WorkspaceSychonizer run into an enless loop when I linked resources into a VirSat project...